cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/bulkwriter/cloudstorage/coordinated/CoordinatedCloudStorageDataTransferApi.java

```diff
@@ -0,0 +1,236 @@
+    {
+        createRestoreSliceInternal(dataTransferApis.getValueOrThrow(clusterId), createSliceRequestPayload);
+    }
+
+    @Override
+    public void restoreJobProgress(RestoreJobProgressFetchPolicy fetchPolicy,
+                                   Predicate<String> clusterIdFilter,
+                                   BiConsumer<String, RestoreJobProgressResponsePayload> progressHandler) throws SidecarApiCallException
+    {
+        forEachInternal(clusterIdFilter, (clusterId, dataTransferApi) -> {
+            restoreJobProgressInternal(clusterId, dataTransferApi, fetchPolicy, progressHandler);
+        });
+    }
+
+    @Nullable
+    @Override
+    public CloudStorageDataTransferApiImpl getValueOrNull(@NotNull String clusterId)
+    {
+        return dataTransferApis.getValueOrNull(clusterId);
+    }
+
+    @Override
+    public void createRestoreJob(CreateRestoreJobRequestPayload createRestoreJobRequestPayload) throws SidecarApiCallException
+    {
+        throw new UnsupportedOperationException("Not supported for coordinated write");
+    }
+
+    @Override
+    public void createRestoreSliceFromExecutor(SidecarInstance sidecarInstance,
+                                               CreateSliceRequestPayload createSliceRequestPayload) throws SidecarApiCallException
+    {
+        throw new UnsupportedOperationException("Not supported for coordinated write");
+    }
+
+    @Override
+    public CompletableFuture<Void> createRestoreSliceFromDriver(SidecarInstance sidecarInstance,
+                                                                CreateSliceRequestPayload createSliceRequestPayload)
+    {
+        throw new UnsupportedOperationException("Not supported for coordinated write");
+    }
+
+    private void createRestoreSliceInternal(CloudStorageDataTransferApiImpl dataTransferApi,
+                                            CreateSliceRequestPayload createSliceRequestPayload) throws SidecarApiCallException
+    {
+        JobInfo jobInfo = dataTransferApi.jobInfo();
+        SidecarClient sidecarClient = dataTransferApi.sidecarClient();
+        QualifiedTableName qualifiedTableName = jobInfo.qualifiedTableName();
+        CreateRestoreJobSliceRequest request = new CreateRestoreJobSliceRequest(qualifiedTableName.keyspace(),
+                                                                                qualifiedTableName.table(),
+                                                                                jobInfo.getRestoreJobId(),
+                                                                                createSliceRequestPayload);
+        RetryPolicy retryPolicy = new CloudStorageDataTransferApiImpl.ExecutorCreateSliceRetryPolicy(sidecarClient);
+        RequestContext requestContext = sidecarClient.requestBuilder().retryPolicy(retryPolicy).request(request).build();
+        try
+        {
+            sidecarClient.executeRequestAsync(requestContext).get();
```

Comment:
you probably don't want this to wait indefinitely, can we upper bound it to something?
```suggestion
            sidecarClient.executeRequestAsync(requestContext).get(1, TimeUnit.MINUTES);
```

Reply:
I would like to avoid the interference from future timing out. 
The request itself has retry policy configured to terminate in a bounded time. I see the sidecar client api desire some refactoring to be more straightforward/intuitive.

Reply:
yeah, that's a fair point. I guess it's not easy to distinguish whether there's a defined upperbound. Maybe something to keep in mind for a v2 of the client API